agent.go

```diff
@@ -0,0 +1,40 @@
+// A simple proxy that multiplexes a unix socket connection
+//
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"syscall"
+
+	"github.com/hyperhq/runv/hyperstart/libhyperstart"
+)
+
+// A new shim agent implementation should have the shimAgent interface
+// Currently it is a subset of libhyperstart.Hyperstart interface but it
+// allows for new agent and protocols.
```

Comment:
I'm not a huge fan of this requirement. I believe the only requirement for an agent should be the gRPC interface. Is that what you meant by `This is based on runv's current Hyperstart interface and can be migrated to the new gRPC APIs once they are settled.` ?

Reply:
This is more like a work around as we have not had an working agent (or mock) for the new gRPC APIs. But we need the shim could be tested.

Reply:
@sameo The shimAgent interface will also be removed once we can import the new gRPC APIs. The idea is not to import runv in shim.go for now. In future, agent.go will not import runv either.

Reply:
Who's responsible for moving gRPC API to kata ? I think we should not merge anything as long as we rely on some runv based packages (even if the content is basically the same.)

Reply:
@sboeuf kata-containers/runtimes#2 is working on this goal.